.github/workflows/asan_test.yml

```diff
@@ -0,0 +1,64 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+name: Address Sanitizer Tests
+
+on:
+  pull_request:
+    paths-ignore:
+    - 'site/**'
+    - 'conan/**'
+    branches:
+    - main
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event_name == 'pull_request' && github.event.number || github.sha }}
+  cancel-in-progress: true
+
+jobs:
+  asan-test:
+    name: "ASAN with ${{ matrix.compiler }} on Ubuntu"
+    runs-on: ubuntu-24.04
```

Comment:
Just a question; Is it a requirement?

Reply:
You mean the ubuntu version? No, I don't think so.

Reply:
Thank you for the answer. Yes, I was wondering if ASAN doesn't work on ubuntu-22.04.

If this is working on ubuntu-22.04, we may want to use `ubuntu-latest` instead of specific versions.

Reply:
BTW, I'm fine for the AS-IS PR too.